recipe/meta.yaml

```
@@ -1,32 +1,38 @@
+{% set name = "bzip2" %}
 {% set version = "1.0.8" %}
 
 package:
-  name: bzip2
+  name: {{ name }}
   version: {{ version }}
 
 source:
-  url: https://sourceware.org/pub/bzip2/bzip2-{{ version }}.tar.gz
+  url: https://sourceware.org/pub/{{ name }}/{{ name }}-{{ version }}.tar.gz
   sha256: ab5a03176ee106d3f0fa90e381da478ddae405918153cca248e682cd0c4a2269
   patches:
-    - 0001-cross.patch
+    - 0001-cross.patch  # [not win]
+    # 2024/2/23: build_dll.patch might not be required in future versions.
+    # Check it before building the next release.
+    - build_dll.patch   # [win]
 
 build:
-  number: 0
+  number: 1
```

```suggestion
  number: 5
```

The highest build number for 1.0.8 is b4 for osx-arm64.

---

Can you check the differences between the recipes in the latest bzip packages available for 1.0.8 and what is in this feedstock?

---

linux-64 <-> linux-aarch64:
```
╰─❯ diff linux-64_bzip2-1.0.8-h7b6447c_0/info/recipe/meta.yaml.template linux-aarch64_bzip2-1.0.8-hfd63f10_2/info/recipe/meta.yaml.template
14c14
<   number: 0
---
>   number: 2
20a21,23
>     - m2-patch  # [win]
>     - m2-gcc-libs  # [win]
>     - patch  # [not win]
```
linux-64 <-> osx-arm64:
about the creation of the osx.arm64's meta.yaml:
```
# This file created by conda-build 3.21.4
# meta.yaml template originally from:
# /Users/ktietz/Code/oss/cf-packages/bzip2-feedstock/recipe, last modified Sun Jun 20 20:52:29 2021
```
```
╰─❯ diff linux-64_bzip2-1.0.8-h7b6447c_0/info/recipe/meta.yaml.template osx-arm64_bzip2-1.0.8-h620ffc9_4/info/recipe/meta.yaml.template
14c14
<   number: 0
---
>   number: 4
59c59
<   license: bzip2
---
>   license: bzip2-1.0.6
71d70
<     - jjhelmus
```

```
diff linux-64_bzip2-1.0.8-h7b6447c_0/info/recipe/bld.bat osx-arm64_bzip2-1.0.8-h620ffc9_4/info/recipe/bld.bat
3a4
> nmake -f makefile.msc
```

```
╰─❯ diff linux-64_bzip2-1.0.8-h7b6447c_0/info/recipe/bld.bat osx-arm64_bzip2-1.0.8-h620ffc9_4/info/recipe/build.sh
1,20c1,19
< REM edited from the one from Anaconda
<
< REM Build step
< jom -f makefile.msc lib bzip2
< if errorlevel 1 exit 1
<
< REM Install step
< copy libbz2.lib %LIBRARY_LIB%\libbz2_static.lib || exit 1
< REM Some packages expect 'bzip2.lib', so make another copy
< copy libbz2.lib %LIBRARY_LIB%\bzip2_static.lib || exit 1
< copy bzlib.h %LIBRARY_INC% || exit 1
<
< cl /O2 /Ibzlib.h /Ibzlib_private.h /D_USRDLL /D_WINDLL blocksort.c bzlib.c compress.c crctable.c decompress.c huffman.c randtable.c /LD /Felibbz2.dll /link /DEF:libbz2.def
< if errorlevel 1 exit 1
<
< copy libbz2.lib %LIBRARY_LIB% || exit 1
< REM Some packages expect 'bzip2.lib', so make copies
< copy libbz2.lib %LIBRARY_LIB%\bzip2.lib || exit 1
< copy libbz2.dll %LIBRARY_BIN% || exit 1
< copy libbz2.dll %LIBRARY_BIN%\bzip2.dll || exit 1
---
> #!/bin/sh
>
> set -ex
>
> # toolchain flags + bzip flags + fpic
> export CFLAGS="${CFLAGS} -Wall -Winline -O2 -g -D_FILE_OFFSET_BITS=64 -fPIC"
> USED_CC=${GCC:-${CC}}
> make install PREFIX=${PREFIX} CFLAGS="${CFLAGS}" CC="${USED_CC}"
>
> # build shared library
> if [[ ${target_platform} =~ .*linux.* ]]; then
>   make -f Makefile-libbz2_so CFLAGS="${CFLAGS}" CC="${USED_CC}"
>   ln -s libbz2.so.${PKG_VERSION} libbz2.so
>   cp -d libbz2.so* ${PREFIX}/lib/
> elif [[ ${target_platform} == osx-* ]]; then
>   ${USED_CC} -shared -Wl,-install_name -Wl,libbz2.dylib -o libbz2.${PKG_VERSION}.dylib blocksort.o huffman.o crctable.o randtable.o compress.o decompress.o bzlib.o ${LDFLAGS}
>   cp libbz2.${PKG_VERSION}.dylib ${PREFIX}/lib/
>   ln -s libbz2.${PKG_VERSION}.dylib ${PREFIX}/lib/libbz2.dylib
> fi
```

I'll reflect on those changes by:
- adding `export CFLAGS="${CFLAGS} -Wall -Winline -O2 -g -D_FILE_OFFSET_BITS=64 -fPIC"` to `build.sh`
- adding `${LDFLAGS}` for `osx-*`

---

I modified the build.sh